### PR TITLE
Manual check command

### LINKED ISSUE #6642: Manual cargo check trigger
I have an override of the cargo check command to run it remotele: 

```
"rust-analyzer.checkOnSave.overrideCommand": [
		"cargo",
		"remote",
		"check",
		"--",
		"--tests",
		"--message-format",
		"json",
	],
```

at the same time, running this on every save is still too much. If check on save is off: 

```
"rust-analyzer.checkOnSave.enable": false,
```

Is there a middle ground where the check command can be triggered manually?

### PR DESCRIPTION
**Pull request by gilescope**
Fixes #6642

There's probably a nicer way to do this. I couldn't see in the typescript side where the checkOnSave was being triggered to do it manually?

### CODE REVIEW

**Comment by bjorn3** [comment]
I think the check is normally not triggered by the extension at all, but by the language server receiving a save event from the editor. I think the command should be implemented in the language server too. The code right now edits global configuration, which affects all windows, not just the current one.

**Comment by gilescope** [comment]
That explains the magic. I'll go looking through previous PRs and see if I can find something vaguely similar where new client-server messages were introduced so I've got a template to follow.

**Comment by gilescope** [comment]
ping: #4604 - Adding startCheck command (no params)

**Comment by gilescope** [comment]
Ok v2 is ready for review @bjorn3

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -690,8 +690,9 @@ impl Config {
             },
         }
     }
-    pub fn flycheck(&self) -> Option<FlycheckConfig> {
-        if !self.data.checkOnSave_enable {
+
+    pub fn flycheck(&self, override_config: bool) -> Option<FlycheckConfig> {
```
Comment (bjorn3) [inline crates/rust-analyzer/src/config.rs:694]: Maybe call it `force_enable`?

File: editors/code/package.json [hunk 2]
```diff
@@ -1349,4 +1358,4 @@
             ]
         }
     }
-}
+}
```
Comment (bjorn3) [inline editors/code/package.json:1361]: nit: missing trailing newline

File: editors/code/src/commands.ts [hunk 3]
```diff
@@ -287,6 +287,18 @@ export function serverVersion(ctx: Ctx): Cmd {
     };
 }
 
+
+export function startCheck(ctx: Ctx): Cmd {
+
```
Comment (bjorn3) [inline editors/code/src/commands.ts:292]:
```suggestion
```

**Comment by bjorn3** [comment]
Couple of nits. Someone else will have to decide if the implementation method is fine.

File: editors/code/package.json (hunk 2 above)
Comment (gilescope) [reply editors/code/package.json:1361]: Thank you so much - just realised there's a vscode extension 'Formatting Toggle'. Am always reaching into the settings to flick it on and off - now I don't have to. Result!

Comment (bjorn3) [reply editors/code/package.json:1361]: You can also use ctrl+shift+i to format the current file once.

**Comment by gilescope** [comment]
Thanks for the quick feedback!

**Comment by gilescope** [comment]
(For people reviewing this - I would seek to do another PR with a stopCheck that halts any in-flight check operation.)

**Comment by gilescope** [comment]
(Squashed)

**Comment by lnicola** [comment]
Closing since this is already available, in a way: https://github.com/rust-analyzer/rust-analyzer/issues/6642#issuecomment-931375360.

**Pull request closed without merging**